drivers/disk/sdmmc_spi.c

```diff
@@ -18,9 +18,11 @@ LOG_MODULE_REGISTER(sdmmc_spi, CONFIG_SDMMC_LOG_LEVEL);
 #include "sdmmc_sdhc.h"
```

Comment:
"Fixes: #32996" is sufficient.

_"...Improves issue reported in #26513, but does not implement CMD6 to
switch to High Speed mode."_ does not belong to the commit message, please remove.

Reply:
>Due to the nature of the SPI API, it is necessary to have two separate
configuration structures to switch clock speed as some SPI drivers only
compare pointers to detect changes.

This is correct behavior that every driver should follow.